website/www/site/content/en/documentation/ml/preprocess-data.md

```diff
@@ -0,0 +1,227 @@
+| ScaleToZScore | See [`tft.scale_to_z_score`](https://www.tensorflow.org/tfx/transform/api_docs/python/tft/scale_to_z_score) in the TensorFlow documentation. |
+| TFIDF | See [`tft.tfidf`](https://www.tensorflow.org/tfx/transform/api_docs/python/tft/tfidf) in the TensorFlow documentation. |:
+{{< /table >}}
+
+You can apply the transforms on either single or multiple columns passed as a
+`dict` on structured data. Keys are column names and values are lists containing
+each column's data.
+
+## I/O requirements {#io}
+
+-   Input to the `MLTransform` class must be in one of the following formats:
+    -   A `dict` of `str`
+        -   Primitive types
+        -   List of primitive types
+        -   NumPy arrays
+-   `MLTransform` outputs a Beam `Row` object with NumPy arrays.
+-   The output `PCollection` is a schema `PCollection`. The output schema
+    contains the transformed columns.
+
+## Artifacts {#artifacts}
+
+Artifacts are additional data elements created by data transformations.
+Examples of artifacts are the minimum and maximum values from a `ScaleTo01`
+transformation, or the mean and variance from a `ScaleToZScore`
+transformation.
+
+In the `MLTransform` class, the `artifact_mode` parameter determines how the
+class handles the artifacts generated or consumed by the data transformations.
+The `artifact_mode` parameter has two states: *produce* and *consume*. The state
+determines whether the `MLTransform` class creates artifacts or retrieves
+artifacts.
+
+### Produce mode {#produce-mode}
+
+When you set `artifact_mode` to `produce`, the `MLTransform` class runs the
+specified transformations on the dataset and then creates artifacts from these
+transformations. The artifacts are stored in the location that you specify in
+the `artifact_location` parameter or in the `MLTransform` output. The default
+value for `artifact_mode` is `produce`.
+
+Produce mode is useful when you want to store the results of your transformations
+for future use. For example, if you apply the same transformations on a
+different dataset, you can use the artifacts to verify whether the
+transformation parameters remain consistent.
+
+The following examples demonstrate how produce mode works.
+
+-   The `ComputeAndApplyVocabulary` transform generates a vocabulary file that contains the
+    vocabulary generated over the entire dataset. The vocabulary file is stored in
+    the location specified by `artifact_location`. The `ComputeAndApplyVocabulary`
+    transform outputs the indices of the vocabulary to the vocabulary file.
+-   The `ScaleToZScore` transform calculates the mean and variance over the entire dataset
+    and then normalizes the entire dataset using the mean and variance. The
+    mean and variance are outputted in the `MLTransform` output.
+    When you set `artifact_mode` to `produce`, these
+    values are stored as a `tensorflow` graph in the location specified by
+    `artifact_location`. You can reuse the values in consume mode.
```

Comment:
`You can reuse the values in consume mode.` - maybe we could be even more explicit here, e.g. `You can reuse the values in consume mode to ensure that a future transformation uses the same mean/variance for normalization`?